servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -270,8 +270,8 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
 
     // Create the final path component. First we need to make sure that there isn't already a file
     // here with that name. If there is an existing file that is a directory and we're creating a
-    // directory, update persistence property of the directories if needed, otherwise, nothing needs
-    // to be done.
+    // directory, update persistence property of the directories if needed, otherwise, throw
```

Comment:
this block of code is likely to be modified---we should not modify. I think your current fix is fine.